src/DataFrames.jl

```diff
@@ -128,4 +128,18 @@ for (dir, filename) in [
     include(joinpath(dir, filename))
 end
 
+# Import  "StringEncodings" if available and set up decoder
+_decoder=nothing
+ENCODING_PACKAGE_READY = false
+function __init__()
+    try Pkg.installed("StringEncodings")
```

Comment:
If we need a dependency on StringEncodings at all it should be a hard dependency. There isn't really a legitimate way to implement optional dependencies right now, though hopefully there will be first-class support for that in the language at some point.